client/src/js/modules/shipment/views/createawb.js

```diff
@@ -138,35 +139,60 @@ define(['backbone',
         _lcFields: ['PHONENUMBER', 'EMAILADDRESS', 'LABNAME', 'ADDRESS', 'CITY', 'POSTCODE','COUNTRY', 'GIVENNAME', 'FAMILYNAME'],
 
         toggleFacilityCourier: function() {
-            const shipmentCountry = this.ui.shipmentCountry.val();
-            if (
-                app.options.get('facility_courier_countries').indexOf(shipmentCountry) > -1
-                || app.options.get('facility_courier_countries_nde').indexOf(shipmentCountry) > -1 
-            ) {
-                if (this.terms.get('ACCEPTED')) {
-                    this.$el.find('.DELIVERYAGENT_AGENTCODE').hide()
-                    this.ui.acc_msg.text('Paid for by Facility')
-                    this.ui.facc.hide()
-                    this.ui.quote.hide()
-                    this.ui.submit.show()
-                    this.ui.termsq.hide()
-                    this.ui.terms.show()
-                    this.shipment.validation.DELIVERYAGENT_AGENTCODE.required = false
-                    this.setPickupDetailsRequired(false)
-                    this.setEmailRequired(false)
-                    this.ui.createAWBForm.hide()
-                    this.ui.submit.text("Proceed")
+
+            if (app.options.get("shipping_service_app_url_incoming")) {
+                this.ui.countrySelect.show()
+                const shipmentCountry = this.ui.shipmentCountry.val();
+                if (
+                    app.options.get('facility_courier_countries').indexOf(shipmentCountry) > -1
+                    || app.options.get('facility_courier_countries_nde').indexOf(shipmentCountry) > -1
+                ) {
+                    if (this.terms.get('ACCEPTED')) {
+                        this.$el.find('.DELIVERYAGENT_AGENTCODE').hide()
+                        this.ui.acc_msg.text('Paid for by Facility')
+                        this.ui.facc.hide()
+                        this.ui.quote.hide()
+                        this.ui.submit.show()
+                        this.ui.termsq.hide()
+                        this.ui.terms.show()
+                        this.shipment.validation.DELIVERYAGENT_AGENTCODE.required = false
+                        this.setPickupDetailsRequired(false)
+                        this.setEmailRequired(false)
+                        this.ui.createAWBForm.hide()
+                        this.ui.submit.text("Proceed")
+                    } else {
+                        this.ui.facc.show()
+                        this.ui.createAWBForm.show()
+                        this.setPickupDetailsRequired(true)
+                        this.setEmailRequired(true)
+                    }
                 } else {
-                    this.ui.facc.show()
+                    this.ui.facc.hide()
```

Comment:
In order for shipments that are paid for by the facility, but aren't suitable for the redirect, I think we need something like this. 
```suggestion
                    if (
                        app.options.get('facility_courier_countries').indexOf(this.lc.get('COUNTRY')) > -1 ||
                        app.options.get('facility_courier_countries_nde').indexOf(this.lc.get('COUNTRY')) > -1
                    ) {
                        this.ui.facc.show()
                    } else {
                        this.ui.facc.hide()
                    }
```

Reply:
That could simplify things. We could then have logic like
```
if (
    app.options.get('facility_courier_countries').indexOf(shipmentCountry) > -1
    && app.options.get("shipping_service_app_url_incoming")
) {
    // Show new form
} else {
    // Show old form
}
```
Where `shipmentCountry` is chosen from a full dropdown.

Reply:
I couldn't simplify it that much, as the "Use facility account" button visibility depends on a few things. But slightly tidier now.